feat: allow to run prepare steps sequentially

KillianHmyd opened the pull request:
This can be useful in the case where we must build a whole package in the prepare step.
In my specific case we use multi-semantic-release to publish some common types and a Docker image. In the prepare step we run a `docker build` in which we need to re-install the dependencies.

Because sometimes the prepare step is not over in the type package the build result in an error because the inlinePlugin started writing half of the package.json to update the dependencies version.

antongolub commented on lib/createInlinePluginCreator.js:
```diff
@@ -181,6 +181,11 @@ function createInlinePluginCreator(packages, multiContext, synchronizer, flags)
 		};
 
 		const prepare = async (pluginOptions, context) => {
+			if (pkg.options.sequentialPrepare) {
```
We're going to separate **semrel** and **msr** flags in future. So let's use `flags.sequentialPrepare` here.

antongolub reviewed (changes requested):
Hey, @KillianHmyd,

could you also add a test for this feature?

antongolub commented on package.json:
```diff
@@ -1,7 +1,7 @@
+	"name": "@payfit/multi-semantic-release",
 	"author": "Dave Houlbrooke <dave@shax.com>",
-	"version": "3.12.4",
+	"version": "3.14.0-rc.2",
```
This should be reverted ) Missed this point.

KillianHmyd commented:
@antongolub Sorry for the package.json, it's reverted 👍 
I've added the test and used the flag instead of the options property.

I've also added a check to be sure that no one use this option while having cyclic dependencies.

antongolub reviewed (approved):
LGTM

antongolub commented:
@KillianHmyd,

It's on the way. Thanks for the contribution.
